packages/modelserver-node/src/client/model-server-client.ts

```diff
@@ -479,7 +479,7 @@ class DefaultTransactionContext implements TransactionContext {
         const result = WebSocketMessageAcceptor.promise<ModelUpdateResult>(this.socket, message => {
```

Comment:
If I read this correctly, the expected type of `result` here is ModelUpdateResult. However, WebSocketMessageAcceptor.promise() returns `message.data`, which contains a message and a patch (It isn't a "ModelUpdateResult").

The problem is that WebSocketMessageAcceptor.promise() takes a Predicate (it returns the raw message data if the Predicate is true, and undefined otherwise), whereas we need a Function (Because we need to convert the raw message data to a ModelUpdateResult - which may be successful or unsuccessful, but shouldn't be 'undefined').

Reply:
🤦 Ah, yes, of course. The new commit e8ed4cbc8f42a53459d32320219205983ee26a5f addresses that.